include/range/v3/utility/functional.hpp

```diff
@@ -295,44 +349,80 @@ namespace ranges
             constexpr auto&& not_ = static_const<not_fn>::value;
         }
 
+        // Lambdas are *not*:
+        // - aggregates             ([expr.prim.lambda]/p3)
+        // - literal types          ([expr.prim.lambda]/p3)
+        // - default constructible  ([expr.prim.lambda]/p20)
+        // - copy assignable        ([expr.prim.lambda]/p20)
+        template<typename Fn>
+        using could_be_lambda_ =
+            meta::bool_<
+               // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=71116
+               //!std::is_literal_type<Fn>::value &&
```

Comment:
> Maybe it's not a compiler bug.

Since the change wasn't a DR, closures should be literal in non-C++1z mode - that's a bug. I'm just pointing out that `is_literal` is not generally useful for detecting lambda-ness.